src/streams.cpp

```diff
@@ -4,14 +4,20 @@
 
 #include <span.h>
 #include <streams.h>
+#include <tinyformat.h>
+#include <util/syserror.h>
 
 #include <array>
 
 std::size_t AutoFile::detail_fread(Span<std::byte> dst)
 {
     if (!m_file) throw std::ios_base::failure("AutoFile::read: file handle is nullptr");
     if (m_xor.empty()) {
-        return std::fread(dst.data(), 1, dst.size(), m_file);
+        const auto n = std::fread(dst.data(), 1, dst.size(), m_file);
+        if (ferror(m_file)) {
+            throw std::ios_base::failure(strprintf("AutoFile::detail_fread: %s", SysErrorString(errno)));
```

Comment:
Alright, moved to the caller.

IMO this is a bugfix, but I am not 100% sure either. Might as well be viewed as "belt-and-suspenders". What is certain is that it can't hurt even if there is never an error on any platform if `detail_fread(dst) == dst.size()`. A unit test would require simulating an IO error under the `fread(3)` call and identically behaving `fread(3)` on all platforms.